### PR TITLE
postcss-nesting: fix pseudo's and allow ampersand everywhere

### PR DESCRIPTION
**Pull request by romainmenke**
see : https://github.com/w3c/csswg-drafts/issues/5745

### CODE REVIEW

File: experimental/postcss-nesting/src/lib/ampersand-to-scope.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+export default function ampersandToScope(rule: Rule) {
+	let parent: Container<Node> = rule.parent;
+
+	while (parent) {
+		if (parent.type === 'rule') {
+			return;
+		}
+
+		parent = parent.parent;
+	}
```
Comment (romainmenke) [inline experimental/postcss-nesting/src/lib/ampersand-to-scope.ts:13]: We want to be sure that `&` was "detected" in a rule that isn't itself nested.

File: experimental/postcss-nesting/src/lib/merge-selectors/merge-selectors.ts [hunk 2]
```diff
@@ -31,12 +31,6 @@ export default function mergeSelectors(node: Node, postcssResult: Result, fromSe
 					return false;
 				}
 
-				// see : https://github.com/w3c/csswg-drafts/issues/7979
-				if (parser.isPseudoElement(maybeCombinator)) {
-					startsWithCombinator = true;
-					return false;
-				}
```
Comment (romainmenke) [inline experimental/postcss-nesting/src/lib/merge-selectors/merge-selectors.ts:38]: I am following what Chrome did and I agree with their analysis : https://github.com/w3c/csswg-drafts/issues/7979#issuecomment-1298337348

**Review by Antonio-Laguna — APPROVED** [review]
This LGTM, should we go minor? Not that it matters a lot but let me know if you disagree

**Comment by romainmenke** [comment]
Minor is good :)

**Pull request merged**